src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -0,0 +1,2458 @@
+    addFormSubHeader(F("Content options"));
+
+    addFormCheckBox(F("Clock with flashing colon"), F("plugin_104_clockflash"), !bitRead(P104_CONFIG_DATETIME, P104_CONFIG_DATETIME_FLASH));
+    addFormCheckBox(F("Clock 12h display"),         F("plugin_104_clock12h"),   bitRead(P104_CONFIG_DATETIME, P104_CONFIG_DATETIME_12H));
+    addFormCheckBox(F("Clock 12h AM/PM indicator"), F("plugin_104_clockampm"),  bitRead(P104_CONFIG_DATETIME, P104_CONFIG_DATETIME_AMPM));
+  }
+  { // Date format
+    const __FlashStringHelper *dateFormats[] = {
+      F("Day Month [Year]"),
+      F("Month Day [Year] (US-style)"),
+      F("[Year] Month Day (Japanese-style)")
+    };
+    const int dateFormatOptions[] = {
+      P104_DATE_FORMAT_EU,
+      P104_DATE_FORMAT_US,
+      P104_DATE_FORMAT_JP
+    };
+    addFormSelector(F("Date format"), F("plugin_104_dateformat"),
+                    3,
+                    dateFormats, dateFormatOptions,
+                    get4BitFromUL(P104_CONFIG_DATETIME, P104_CONFIG_DATETIME_FORMAT));
+  }
+  { // Date separator
+    const __FlashStringHelper *dateSeparators[] = {
+      F("Space"),
+      F("Slash /"),
+      F("Dash -"),
+      F("Dot <b>.</b>")
+    };
+    const int dateSeparatorOptions[] = {
+      P104_DATE_SEPARATOR_SPACE,
+      P104_DATE_SEPARATOR_SLASH,
+      P104_DATE_SEPARATOR_DASH,
+      P104_DATE_SEPARATOR_DOT
+    };
+    addFormSelector(F("Date separator"), F("plugin_104_dateseparator"),
+                    4,
+                    dateSeparators, dateSeparatorOptions,
+                    get4BitFromUL(P104_CONFIG_DATETIME, P104_CONFIG_DATETIME_SEP_CHAR));
+
+    addFormCheckBox(F("Year uses 4 digits"), F("plugin_104_year4dgt"), bitRead(P104_CONFIG_DATETIME, P104_CONFIG_DATETIME_YEAR4DGT));
+  }
+  # endif // ifdef P104_USE_DATETIME_OPTIONS
+
+  addFormSubHeader(F("Zones"));
+
+  { // Zones
+    String zonesList[P104_MAX_ZONES];
+    int    zonesOptions[P104_MAX_ZONES];
+
+    for (uint8_t i = 0; i < P104_MAX_ZONES; i++) {
+      zonesList[i]    = i + 1;
+      zonesOptions[i] = i + 1; // No 0 needed or wanted
+    }
+    # if defined(P104_USE_TOOLTIPS) || defined(P104_ADD_SETTINGS_NOTES)
+    String zonetip;
+
+    if (zonetip.reserve(90)) {
+      zonetip  = F("Select between 1 and ");
+      zonetip += P104_MAX_ZONES;
+      zonetip += F(" zones, changing");
+      #  ifdef P104_USE_ZONE_ORDERING
+      zonetip += F(" Zones or Zone order");
+      #  endif // ifdef P104_USE_ZONE_ORDERING
+      zonetip += F(" will save and reload the page.");
+    }
+    # endif    // if defined(P104_USE_TOOLTIPS) || defined(P104_ADD_SETTINGS_NOTES)
+    addFormSelector(F("Zones"), F("plugin_104_zonecount"), P104_MAX_ZONES, zonesList, zonesOptions, NULL, P104_CONFIG_ZONE_COUNT, true
+                    # ifdef P104_USE_TOOLTIPS
+                    , zonetip
+                    # endif // ifdef P104_USE_TOOLTIPS
+                    );
+
+    # ifdef P104_USE_ZONE_ORDERING
+    const String orderTypes[] = {
```

Comment:
This one can be of type flash string.
`const __FlashStringHelper * orderTypes[] = {`

Reply:
There is no overload for `addFormSelector()` (yet) that accepts that, combined with a tooltip argument.

Reply:
OK, it only is about 2 strings here, so no big deal.